packages/flutter/lib/src/services/text_input.dart

```diff
@@ -14,6 +14,7 @@ import 'dart:ui' show
   hashValues;
 
 import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
```

Comment:
I think this causes a dependency loop, since material/text_field.dart also imports services.dart.

Reply:
I'm trying to fix the circular ref. Here is my proposal:

Split pull the member not related to the selection toolbar into an abstract class called `TextEditingDelegate` since it pretty much just contains the `TextEditingValue`.

Then move `TextSelectionDeletate` to widgets/text_selection.dart, and make it inherit from TextEditingDelegate.

This fixes the circular ref, but I wanted to hear what you thought about it. @justinmc 

I've coded it up and it works.

Reply:
Will get back to you on this after working on the design doc I mentioned below.